### PR TITLE
feat: added authentication

### PR DESCRIPTION
**Pull request by himadri2110**
- Login 
- Sign Up
- Logout

- Dummy home page

### CODE REVIEW

**Review by BelideNikhil — APPROVED** [review]
The code structure is well sorted!
Nice work on toolkit error handling part!

Good to merge

**Review by srishti-maurya — APPROVED** [review]
Code is clean, well structured and readable.
Great work 👍

File: public/index.html [hunk 1]
```diff
@@ -2,13 +2,16 @@
...
+    <meta name="description" content="Sapphire - A Social Media App" />
+
+    <!-- Font Awesome -->
+    <script
+      src="https://kit.fontawesome.com/0e889649eb.js"
+      crossorigin="anonymous"
+    ></script>
 
     <!--
       manifest.json provides metadata used when your web app is installed on a
```
Comment (gopalbbharadva) [inline public/index.html:17]: You can remove these default comments

File: public/index.html [hunk 2]
```diff
@@ -2,13 +2,16 @@
 <html lang="en">
   <head>
     <meta charset="utf-8" />
-    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <link rel="icon" href="%PUBLIC_URL%/sapphire.svg" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
     <meta name="theme-color" content="#000000" />
```
Comment (gopalbbharadva) [inline public/index.html:7]: Suggestion: You can give your primary color hex code here.

**Review by gopalbbharadva — APPROVED** [review]
Feature working fine. good to go.

File: public/index.html (hunk 2 above)
Comment (himadri2110) [reply public/index.html:7]: Yes, thank you. I'll update it.

**Pull request merged**